.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,74 @@
+name: Build & Test
+
+on:
+  push:
+    branches: ["*"]
+
+env:
+  # Cross-compilation for aarch64 requires a different linker
+  CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER: aarch64-linux-gnu-gcc
+
+permissions:
+  contents: read
+
+jobs:
+  Tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - x86_64-unknown-linux-gnu
+          - aarch64-unknown-linux-gnu
+          - x86_64-pc-windows-msvc
+          - x86_64-apple-darwin
+          - aarch64-apple-darwin
```

Comment:
Do we need to run tests on different archs for the same OS?

Reply:
GitHubs Action matrix is a little counter-intuitive. A job for each `target` (currently 5) and each `rustup_toolchain` (currently 1) is created. The `include` part sets the os. We use the native os for each target, except linux arm. I thought testing on the native arch is best when ever possible - but I know very litte about the rust build standards.

If you think testing with cross compiling is better or also needed required let me know the exact test matrix.